apps/nextjs/src/components/reviews.tsx

```
@@ -79,8 +131,33 @@ export function CreateReviewForm() {
     },
   });
 
-  const onSubmit = (data: z.input<typeof ReviewFormSchema>) => {
-    createReview.mutate(CreateReviewSchema.parse(data));
+  const onSubmit = async (data: z.input<typeof ReviewFormSchema>) => {
+    const files: Blob[] = data.files;
+    const urls: string[] = [];
+    const sizes: number[] = [];
+    // Upload every file from the form
+    for (const file of files) {
+      try {
+        const res = await uploadFile(file);
+        urls.push(res.url);
+        sizes.push(file.size);
+      } catch (error) {
+        toast.error("Failed to upload photos");
+        console.error(error);
+        return;
```

does returning here work with the "submit" handling? do you need to throw an error instead?